### ISSUE TITLE
Implementing containerSecurityContext.seccompProfile breaks compatibility with PodSecurityPolicy

### ISSUE THREAD

**Issue by rptaylor** [bitnami/charts#21510]
### Name and Version

bitnami/node-exporter 3.9.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

On a k8s v1.23 cluster with PodSecurityPolicy enabled, install node-exporter 3.9.0 or newer (included in kube-prometheus versions after 8.21.6)

`helm -n kube-prometheus --version=8.22.2 --repo=https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami  install kube-prometheus kube-prometheus`

### Are you using any custom parameters or values?

No

### What is the expected behavior?

The node-exporter daemonset should run abiding by the built-in kube-prometheus-node-exporter-kube-prometheus PSP that gets created and set up.

```
$ kubectl -n kube-prometheus get daemonsets.apps kube-prometheus-node-exporter
NAME                            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
kube-prometheus-node-exporter   3         3         3       3            3           <none>          2m31s
```

### What do you see instead?

The pods fail to run due to PSP violations:
```
54s         Warning   FailedCreate             daemonset/kube-prometheus-node-exporter                    Error creating: pods "kube-prometheus-node-exporter-" is forbidden: PodSecurityPolicy: unable to admit pod: [spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed to be used spec.securityContext.hostPID: Invalid value: true: Host PID is not allowed to be used spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.volumes[1]: Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].hostPort: Invalid value: 9100: Host port 9100 is not allowed to be used. Allowed ports: [] pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/node-exporter]: Forbidden: seccomp may not be set]
```

### Additional information

Commits such as https://github.com/bitnami/charts/pull/20514 for https://github.com/bitnami/charts/issues/20000 break compatibility with PodSecurityPolicy.
This is because AFAICT  "RuntimeDefault" is not a valid seccomp profile according to the PodSecurityPolicy API. It may only be valid for PSA, not PSP.  If I try to set the PSP annotation  `seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'RuntimeDefault'`  (which is how seccomp is controlled in PSP), it fails:
```
# podsecuritypolicies.policy "kube-prometheus-node-exporter-kube-prometheus" was not valid:
# * metadata.annotations[seccomp.security.alpha.kubernetes.io/allowedProfileNames]: Invalid value: "RuntimeDefault": must be a valid seccomp profile
```

Instead, the default seccomp profile as understood by PSP is called 'runtime/default'.

So then I tried setting the PSP annotation to allow the 'runtime/default' seccomp profile, which is accepted as valid.
However if I try to modify the seccomp profile used by the node-exporter daemonset to also use 'runtime/default', it is rejected:
```
DaemonSet.apps \"kube-prometheus-node-exporter\" is invalid: spec.template.spec.containers[0].securityContext.seccompProfile.type: Unsupported value: \"runtime/default\": supported values: \"Localhost\", \"RuntimeDefault\", \"Unconfined\
```

So there is no valid value of seccomp profile that is accepted by PSP and also matches the daemonset.
The only way to avoid this is to not have any seccomp profile set in the containerSecurityContext of the node-exporter daemonset at all. However, due to longstanding Helm bugs (https://github.com/helm/helm/issues/9027 , https://github.com/helm/helm/issues/12488 ) it is not possible to remove the value by setting `containerSecurityContext.seccompProfile.type` to null.

That being the case there is no way to run versions of node-exporter newer than 3.8.5 on a cluster using PodSecurityPolicy (at least with k8s 1.23).
I understand PSP is deprecated but it will not be removed from kubernetes until v1.25 and some people may still be using it.

**Comment by tspearconquest** [comment]
Try "runtime/default" instead of "RuntimeDefault"

IIRC, the older kube versions before seccomp was GA used the former.

**Comment by rptaylor** [comment]
> Try "runtime/default" instead of "RuntimeDefault"

As I mentioned in the Additional information section this does not seem to work either.

**Comment by tspearconquest** [comment]
Hi, I edited my post to be more clear, as I did see that. Try adding it where you did not yet add it.

**Comment by tspearconquest** [comment]
https://web.archive.org/web/20220402043022/https://kubernetes.io/docs/concepts/security/pod-security-policy/#seccomp

Here is a Web Archive copy of the kubernetes PSP docs indicating that `runtime/default` is the correct value to use for the PSP annotation.

**Comment by rptaylor** [comment]
It was not explicit in my post (edited) but I did do that before trying to set the node exporter daemonset seccomp profile to 'runtime/default' (which is rejected as invalid). 

But I might have missed the possibility that the PSP and daemonset interpret runtime/default and RuntimeDefault, respectively, as the same thing, so the node exporter seccomp profile should have remained RuntimeDefault.
So I tried that too, and it did not work.

BUT I finally found that if `seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default` is also set along with `seccomp.security.alpha.kubernetes.io/allowedProfileNames: runtime/default` and the daemonset seccomp profile remains unchanged, it works!

However unfortunately there is no way to set annotations for the PSP specifically, there is only commonAnnotations which would apply to all objects, that would be a bit of a hack.


https://github.com/bitnami/charts/blob/main/bitnami/node-exporter/values.yaml#L38

https://github.com/bitnami/charts/blob/main/bitnami/node-exporter/templates/psp.yaml#L14
